crates/client/src/lib.rs

```diff
@@ -533,11 +619,59 @@ impl<R: RegistryStorage, C: ContentStorage> Client<R, C> {
                     packages: Cow::Borrowed(&last_known),
                 })
                 .await
-                .map_err(|e| {
-                    ClientError::translate_log_not_found(e, |id| {
+            {
+                Ok(res) => res,
+                Err(api::ClientError::LogNotFoundWithHint(log_id, header)) => {
```

Comment:
This kind of user interaction logic probably doesn't belong in this client library, at least not in this low-level method. The simplest place to put it would be in the warg bin code or - if it needs to be reused by other dependants - at least in a separate method or gated behind an opt-in flag of some sort.